packages/flutter/lib/src/services/text_editing.dart

```diff
@@ -144,4 +145,106 @@ class TextSelection extends TextRange {
       isDirectional: isDirectional ?? this.isDirectional,
     );
   }
+
+  /// Return the selection expanded to the given [TextPosition].
+  ///
+  /// If the given [TextPosition] is inside of the selection, then returns
+  /// `this` without change.
+  ///
+  /// The returned selection will always be a strict superset of the current
+  /// selection. In other words, the selection grows to include the given
+  /// [TextPosition].
+  ///
+  /// If extentAtIndex is true, then the [TextSelection.extentOffset] will be
+  /// placed at the given index regardless of the original order of it and
+  /// [TextSelection.baseOffset]. Otherwise, their order will be preserved.
+  ///
+  /// ## Difference with [extendTo]
+  /// In contrast with this method, [extendTo] is a pivot; it holds
+  /// [TextSelection.baseOffset] fixed while moving [TextSelection.extentOffset]
+  /// to the given [TextPosition].  It doesn't strictly grow the selection and
+  /// may collapse it or flip its order.
+  TextSelection expandTo(TextPosition position, [bool extentAtIndex = false]) {
+    final int upperOffset = math.min(baseOffset, extentOffset);
```

Comment:
This is `end`. and `lowerOffset` is `start`.

Reply:
nit: the implementation is a bit difficult to read.

Reply:
Agreed.  I cleaned this up a bit and added some comments.